File: solutions/fsharp/bob/1/Bob.fs
module Bob

open System

let response (input: string) : string =
    match input.Trim() with
    | r when r |> String.IsNullOrWhiteSpace -> "Fine. Be that way!"
    | r when
        r.EndsWith '?'
        && (r
            |> Seq.filter (fun c -> c |> Char.IsLetter)
            |> fun filtered -> not (Seq.isEmpty filtered) && Seq.forall Char.IsUpper filtered)
        ->
        "Calm down, I know what I'm doing!"
    | r when
        r
        |> Seq.filter (fun c -> c |> Char.IsLetter)
        |> fun filtered -> not (Seq.isEmpty filtered) && Seq.forall Char.IsUpper filtered
        ->
        "Whoa, chill out!"
    | r when r.EndsWith '?' -> "Sure."
    | _ -> "Whatever."

File: solutions/csharp/bob/1/Bob.cs
using System;
using System.Linq;

public static class Bob
{
    public static string Response(string statement)
    {
        statement = statement.Trim();
        if (string.IsNullOrEmpty(statement)) return "Fine. Be that way!";

        var upperCase = statement.ToUpper();
        if (statement.Any(char.IsLetter))
        {
            if (statement.Equals(upperCase) && statement[statement.Length - 1] == '?')
                return "Calm down, I know what I'm doing!";
            else if (statement.Equals(upperCase))
                return "Whoa, chill out!";
            else if (statement[statement.Length - 1] == '?')
                return "Sure.";
        }
        else if (statement[statement.Length - 1] == '?')
            return "Sure.";

        return "Whatever.";
    }
}